ios/RNLinksdk.m

```diff
@@ -93,107 +88,85 @@ - (void)stopObserving {
             [strongSelf sendEventWithName:kRNLinkKitOnEventEvent
-                                     body:@{kRNLinkKitEventNameKey: event, kRNLinkKitEventMetadataKey: metadata}];
+                                     body:@{kRNLinkKitEventNameKey: event.eventName, kRNLinkKitEventMetadataKey: eventMetadata}];
         }
     };
 
-    if ([linkTokenInput length] > 0) {
-        if (isUsingLinkToken) {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithLinkToken:linkTokenInput
-                                                                               oauthStateId:oauthStateId
-                                                                              configuration:linkConfiguration
-                                                                                   delegate:self.linkViewDelegate];
-        }
-        else if ([linkTokenInput hasPrefix:kRNLinkKitItemAddTokenPrefix]) {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithItemAddToken:linkTokenInput
-                                                                                 configuration:linkConfiguration
-                                                                                      delegate:self.linkViewDelegate];
-        }
-        else {
-            self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithPublicToken:linkTokenInput
-                                                                                configuration:linkConfiguration
-                                                                                     delegate:self.linkViewDelegate];
-        }
-    }
-    else if ([institution length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithInstitution:institution
-                                                                            configuration:linkConfiguration
-                                                                                 delegate:self.linkViewDelegate];
-    }
-    else if ([paymentTokenInput length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithPaymentToken:paymentTokenInput
-                                                                              oauthStateId:oauthStateId
-                                                                             configuration:linkConfiguration
-                                                                                  delegate:self.linkViewDelegate];
-    }
-    else if ([oauthStateId length] > 0) {
-        self.linkViewController = [[PLKPlaidLinkViewController alloc] initWithOAuthStateId:oauthStateId
-                                                                             configuration:linkConfiguration
-                                                                                  delegate:self.linkViewDelegate];
+    if (isUsingLinkToken) {
```

Comment:
Minor difference we check for public key first and try to open with that.  If there is no public key then we create a link token configuration.  I think in most cases it'll be about the same except when you provide a public key AND link token. On iOS it will open with a token config ignoring the key, on android it will open with a public key config with a link token (I'm actually not sure off the top of my head what happens then)

Reply:
on Android it should open with the link-token since the server discards the public key value

Reply:
It sounds like from a configuration standpoint it's different (i.e. we end up constructing a LinkTokenConfiguration whereas Android constructs a LinkPublicKeyConfiguration with a link token), but I believe the behaviour will be the same (as Amy mentioned the server discards the public key value). IMHO it makes more sense to have Link Token have more precedence, and maybe log to the console a warning if both are set